### PR TITLE
Depth of field optimizations and cleanup

### PR DESCRIPTION
**Pull request by kircher1**
The main change here is to use R-only texture format for the circle of confusion (which only needs to store a single value per pixel) rather than using an RGBA texture. This will save space and should be more bandwidth efficient to sample from during the subsequent blur passes.

Also removes `sampleDistance` from the blur shader-- it's not used AFAICT and deleting this code removes the need for passing cameraMinMaxZ uniform.

Other changes here are just minor cleanup.

### CODE REVIEW

**Review by sebavan — APPROVED** [review]
Thanks a lot

File: packages/dev/core/src/PostProcesses/blurPostProcess.ts [hunk 1]
```diff
@@ -106,14 +106,15 @@ export class BlurPostProcess extends PostProcess {
         samplingMode: number = Texture.BILINEAR_SAMPLINGMODE,
         engine?: Engine,
         reusable?: boolean,
-        textureType: number = Constants.TEXTURETYPE_UNSIGNED_INT,
+        textureType = Constants.TEXTURETYPE_UNSIGNED_INT,
         defines = "",
-        private _blockCompilation = false
+        private _blockCompilation = false,
+        textureFormat = Constants.TEXTUREFORMAT_RGBA
```
Comment (Popov72) [inline packages/dev/core/src/PostProcesses/blurPostProcess.ts:112]: You need to document `textureFormat` in the function doc above.

File: packages/dev/core/src/PostProcesses/depthOfFieldBlurPostProcess.ts [hunk 2]
```diff
@@ -57,11 +57,12 @@ export class DepthOfFieldBlurPostProcess extends BlurPostProcess {
...
         imageToBlur: Nullable<PostProcess> = null,
-        samplingMode: number = Texture.BILINEAR_SAMPLINGMODE,
+        samplingMode = Texture.BILINEAR_SAMPLINGMODE,
         engine?: Engine,
         reusable?: boolean,
-        textureType: number = Constants.TEXTURETYPE_UNSIGNED_INT,
-        blockCompilation = false
+        textureType = Constants.TEXTURETYPE_UNSIGNED_INT,
+        blockCompilation = false,
+        textureFormat = Constants.TEXTUREFORMAT_RGBA
```
Comment (Popov72) [inline packages/dev/core/src/PostProcesses/depthOfFieldBlurPostProcess.ts:65]: Same comment than above.

**Review by Popov72 — APPROVED** [review]
Nice!

**Pull request merged**